tools/nightly.py

```diff
@@ -75,7 +86,20 @@
     Version = None  # type: ignore[assignment,misc]
 
 
-REPO_ROOT = Path(__file__).absolute().parent.parent
+@functools.lru_cache
+def _find_repo_root() -> Path:
+    """Find the root of the git repository where this script is located."""
+    script_dir = Path.cwd().absolute()
+    result = subprocess.run(
+        ["git", "rev-parse", "--show-toplevel"],
+        cwd=script_dir,
+        capture_output=True,
+        text=True,
+        check=True,
+    )
+    return Path(result.stdout.strip()).absolute()
```

Comment:
nit: use `check_output` and add a failback

```suggestion
@functools.cache
def _find_repo_root() -> Path:
    """Find the root of the git repository where this script is located."""
    script_dir = Path.cwd().absolute()
    try:
        repo_dir = subprocess.check_output(
            ["git", "rev-parse", "--show-toplevel"],
            stderr=subprocess.DEVNULL,
            cwd=script_dir,
            text=True,
            encoding="utf-8",
        )
    except subprocess.CalledProcessError:
        return Path(__file__).absolute().parent.parent
    return Path(repo_dir.strip()).absolute()
```

Reply:
3.9+ !